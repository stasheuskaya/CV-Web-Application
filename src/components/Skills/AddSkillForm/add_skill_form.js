import '../skills.css';
import { Formik, Field, Form, ErrorMessage } from 'formik';
import * as Yup from 'yup';
import { useDispatch } from "react-redux";
import { saveSkill } from '../../../store/thunks/thunks';

const initialValues = {
    skillName: '',
    skillRange: '',
}

export default function AddSkillForm() {
  const dispatch = useDispatch();

    return (
        <div className = 'addSkillForm'>
          <Formik
            initialValues = {initialValues}
            validationSchema = {Yup.object({
              skillName: Yup.string()
              .required('Skill name is a required field'),
              skillRange: Yup.number()
              .min(10, 'Skill range must be greater than or equal to 10')
              .required('Skill range is a required field')
              .max(100, 'Skill range must be less than or equal to 100')
            })}
            onSubmit = {(values, { resetForm }) => {
              const skill = {name: values.skillName, range: values.skillRange}
              dispatch(saveSkill(skill));
              resetForm();
            }}
          >
            {({ errors, isValid }) => (
              <Form>
                <div className='skillName'>
                  <h2><label htmlFor="skillName">Skill Name</label></h2>
                  <Field className={`${errors.skillName ? "errBorder" : ""}`} id="skillName" name="skillName" placeholder="Enter skill name" />
                </div>
                <div className='inputErrorMessage'><ErrorMessage name="skillName" /></div>
                <div className='skillRange'>
                  <h2><label htmlFor="skillRange">Skill range</label></h2>
                  <Field className={`${errors.skillRange ? "errBorder" : ""}`} id="skillRange" name="skillRange" placeholder="Enter skill range" />
                </div>
                <div className='inputErrorMessage'><ErrorMessage name="skillRange" /></div>
                <div className='addSkillButton'>
                  <button type="submit" disabled={!isValid}>Add Skill</button>
                </div>
              </Form>
            )}
          </Formik>
        </div>
    )
}