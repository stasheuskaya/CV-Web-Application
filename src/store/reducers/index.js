import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;