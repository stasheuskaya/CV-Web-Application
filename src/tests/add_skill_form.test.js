import renderer from "react-test-renderer";
import AddSkillForm from "../components/Skills/AddSkillForm/add_skill_form";
import { Provider } from 'react-redux';
import { store } from "../store/store"

describe('AddSkillForm Page', () => {
    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
                <AddSkillForm />
			</Provider>)).toMatchSnapshot();
	});
});