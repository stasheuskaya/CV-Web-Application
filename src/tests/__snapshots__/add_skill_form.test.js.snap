// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSkillForm Page renders content 1`] = `
<div
  className="addSkillForm"
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="skillName"
    >
      <h2>
        <label
          htmlFor="skillName"
        >
          Skill Name
        </label>
      </h2>
      <input
        className=""
        id="skillName"
        name="skillName"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter skill name"
        value=""
      />
    </div>
    <div
      className="inputErrorMessage"
    />
    <div
      className="skillRange"
    >
      <h2>
        <label
          htmlFor="skillRange"
        >
          Skill range
        </label>
      </h2>
      <input
        className=""
        id="skillRange"
        name="skillRange"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter skill range"
        value=""
      />
    </div>
    <div
      className="inputErrorMessage"
    />
    <div
      className="addSkillButton"
    >
      <button
        disabled={false}
        type="submit"
      >
        Add Skill
      </button>
    </div>
  </form>
</div>
`;
